scene/3d/skeleton_3d.cpp

```diff
@@ -145,47 +170,29 @@ void Skeleton3D::_update_process_order() {
 	Bone *bonesptr = bones.ptrw();
 	int len = bones.size();
 
-	process_order.resize(len);
-	int *order = process_order.ptrw();
+	parentless_bones.clear();
+
 	for (int i = 0; i < len; i++) {
 		if (bonesptr[i].parent >= len) {
 			//validate this just in case
 			ERR_PRINT("Bone " + itos(i) + " has invalid parent: " + itos(bonesptr[i].parent));
 			bonesptr[i].parent = -1;
 		}
-		order[i] = i;
-		bonesptr[i].sort_index = i;
-	}
-	//now check process order
-	int pass_count = 0;
-	while (pass_count < len * len) {
-		//using bubblesort because of simplicity, it won't run every frame though.
-		//bublesort worst case is O(n^2), and this may be an infinite loop if cyclic
-		bool swapped = false;
-		for (int i = 0; i < len; i++) {
-			int parent_idx = bonesptr[order[i]].parent;
-			if (parent_idx < 0) {
-				continue; //do nothing because it has no parent
-			}
-			//swap indices
-			int parent_order = bonesptr[parent_idx].sort_index;
-			if (parent_order > i) {
-				bonesptr[order[i]].sort_index = parent_order;
-				bonesptr[parent_idx].sort_index = i;
-				//swap order
-				SWAP(order[i], order[parent_order]);
-				swapped = true;
-			}
-		}
+		bonesptr[i].child_bones.clear();
```

Comment:
Doing the process in just one loop and removing the child bones for each bone is causing a regression when the bones are not sorted from root to leaves (if the root is last, all other bones are missing, since it removes its child bones).

Adding a first loop just for cleaning the child bones fixes the issue, but I wonder if there's a better solution (like sorting the bones beforehand).

You can reproduce the issue in @TokageItLab's project here: https://github.com/TokageItLab/3d-platform-test-for-godot4